src/org/labkey/test/components/ui/grids/EditableGrid.java

```diff
@@ -595,9 +611,51 @@ public EditableGrid pasteFromCell(int row, String columnName, String pasteText)
         // ... or for a second and a half
         WebDriverWrapper.waitFor(()-> (getRowCount() > initialRowCount || !indexValue.equals(gridCell.getText())) &&
                         isInSelection(gridCell), 1500);
+        if (validate)
+            waitForAnyPasteContent(pasteText);
+
         return this;
     }
 
+    /**
+     * Awaits any elements (except for empty, or space-only) of the pasted content to be present in the grid
+     * @param pasteContent  tab-separated text
+     */
+    protected void waitForAnyPasteContent(String pasteContent)
+    {
+        // split pasteContent into its parts
+        var contentParts = pasteContent.replace("\n", "\t").split("\t");
+        // filter out empty and space-only values
+        var filteredParts = Arrays.stream(contentParts).filter(a-> !a.isEmpty() && !a.equals(" ")).collect(Collectors.toList());
+        await().atMost(Duration.ofSeconds(2))
+                .untilAsserted(()-> Assertions.assertThat(getSelectionCellTexts())
+                        .containsAnyElementsOf(filteredParts));
+    }
+
+    /**
+     * Awaits all elements (except empty or space-only) of the pasted content to be present in the grid.
+     * Use this to validate all expected content appears after pasting to the grid
+     * @param pasteContent  tab-separated text of the sort usually pasted into the edit grid
+     */
+    public void waitForPasteContent(String pasteContent)
+    {
+        // split pasteContent into its parts
+        var contentParts = pasteContent.replace("\n", "\t").split("\t");
+        // filter out empty and space-only values
+        var filteredParts = Arrays.stream(contentParts).filter(a-> !a.isEmpty() && !a.equals(" ")).collect(Collectors.toList());
+        await().atMost(Duration.ofSeconds(2))
+                .untilAsserted(()-> Assertions.assertThat(getSelectionCellTexts())
+                        .containsAll(filteredParts));
+    }
```

Comment:
It doesn't look like this method is used.
This should not be public.

Reply:
It is offered as an option to changing the behavior of `pasteFromCell`; if it's public callers can chain this method after pasting content if they want to assert that content.
Please, if you're going to press for no change and then press for no new public methods, consider that both of these options at once doesn't make sense.  If the method is private and not used internally, it might as well not exist and the purpose for this whole PR is... useless

If you want to tell me what to do with the code, please let's talk about it instead of doing this in text